File: src/main/java/io.fedcuit.github/futurePattern/RealData.java
package io.fedcuit.github.futurePattern;

import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class RealData implements Data<String> {

    private final String result;

    RealData(final String param) {
        this.result = IntStream.range(0, 10).mapToObj(i -> {
            try {
                Thread.sleep(100);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
            return param;
        }).collect(Collectors.joining());
    }

    @Override
    public String get() {
        return this.result;
    }
}

File: src/main/java/io.fedcuit.github/futurePattern/RealData.kt
package io.fedcuit.github.futurePattern

class RealData(param: String) : Data<String> {
    private var result: String

    init {
        this.result = (0..10).joinToString {
            try {
                Thread.sleep(100)
            } catch (e: Exception) {
                e.printStackTrace()
            }; param
        }
    }


    override fun get(): String = this.result

}